### PR TITLE
`VTK` and `find-superquadric` delegated to `superbuild`

### PR DESCRIPTION
**Pull request by pattacini**
Install `vtk` and `find-superquadric` via `superbuild`.
See https://github.com/robotology/robotology-superbuild/releases/tag/v2022.09.0.

### CODE REVIEW

**Comment by pattacini** [comment]
Hi @traversaro 

I stumbled upon the following error while building `find-superquadric` w/ 2022.09.0:

```
-- Found ICUB 1.28.1 (Components: iCubDev ctrlLib skinDynLib iKin iDyn learningMachine perceptiveModels actionPrimitives optimization)
2022-09-29T09:42:28.2189418Z [91mCMake Error at CMakeLists.txt:28 (find_package):
2022-09-29T09:42:28.2189844Z   By not providing "FindVTK.cmake" in CMAKE_MODULE_PATH this project has
2022-09-29T09:42:28.2190235Z   asked CMake to find a package configuration file provided by "VTK", but
2022-09-29T09:42:28.2190541Z   CMake did not find one.
2022-09-29T09:42:28.2190687Z 
2022-09-29T09:42:28.2190867Z   Could not find a package configuration file provided by "VTK" with any of
2022-09-29T09:42:28.2191157Z   the following names:
2022-09-29T09:42:28.2191299Z 
2022-09-29T09:42:28.2191391Z     VTKConfig.cmake
2022-09-29T09:42:28.2191665Z     vtk-config.cmake
2022-09-29T09:42:28.2191803Z 
2022-09-29T09:42:28.2191982Z   Add the installation prefix of "VTK" to CMAKE_PREFIX_PATH or set "VTK_DIR"
2022-09-29T09:42:28.2192373Z   to a directory containing one of the above files.  If "VTK" provides a
2022-09-29T09:42:28.2192733Z   separate development package or SDK, be sure it has been installed.
2022-09-29T09:42:28.2192941Z 
2022-09-29T09:42:28.2192946Z 
2022-09-29T09:42:28.2193151Z [0m-- Configuring incomplete, errors occurred!
```

Ref: https://github.com/robotology/icub-gazebo-grasping-sandbox/actions/runs/3149955325/jobs/5122178030.

Do you have any clue?

**Comment by traversaro** [comment]
I did not add pcl and vtk to `install_apt_dependencies.sh` given their size and the fact that they are not supported on 18.04 (a bit like Gazebo). Installing manually `libpcl-dev` should solve the problem.

**Comment by pattacini** [comment]
Thanks 👍🏻 
I'll try it out.

File: dockerfiles/Dockerfile [hunk 1]
```diff
@@ -37,7 +37,10 @@ RUN apt install -y magic-wormhole
 RUN git clone https://github.com/novnc/noVNC.git /opt/novnc && \
     git clone https://github.com/novnc/websockify /opt/novnc/utils/websockify && \
     echo "<html><head><meta http-equiv=\"Refresh\" content=\"0; url=vnc.html?autoconnect=true&reconnect=true&reconnect_delay=1000&resize=scale&quality=9\"></head></html>" > /opt/novnc/index.html
-   
+
+# Install PCL and VTK
```
Comment (pattacini) [inline dockerfiles/Dockerfile:41]:
```suggestion
# Install PCL and VTK (a dependency for PCL)
```

**Review by traversaro — APPROVED** [review]

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -7,12 +7,16 @@
 
 find_package(YARP REQUIRED COMPONENTS os dev sig math)
 find_package(ICUB REQUIRED COMPONENTS iKin)
+
+set(CMAKE_MODULE_PATH_BAK ${CMAKE_MODULE_PATH})
+find_package(Gazebo REQUIRED)
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH_BAK})
+
```
Comment (pattacini) [inline src/CMakeLists.txt:14]: This modification is required to get around conflicts between VTK and GAZEBO in Ubuntu 2022.04.
Thanks @traversaro for the patch!

Comment (pattacini) [inline src/CMakeLists.txt:14]:
```suggestion

# Gte around conflicts between VTK and Gazebo in Ubuntu 2022.04
set(CMAKE_MODULE_PATH_BAK ${CMAKE_MODULE_PATH})
find_package(Gazebo REQUIRED)
set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH_BAK})

```

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -7,12 +7,17 @@
 
 find_package(YARP REQUIRED COMPONENTS os dev sig math)
 find_package(ICUB REQUIRED COMPONENTS iKin)
+
+# Gte around conflicts between VTK and Gazebo in Ubuntu 2022.04
```
Comment (pattacini) [inline src/CMakeLists.txt:11]:
```suggestion
# Get around conflicts between VTK and Gazebo in Ubuntu 2022.04
```

**Pull request merged**